tools/issue-labeler/src/IssueLabelerService/Answers/OpenAiAnswerService.cs

```diff
@@ -18,102 +19,151 @@ public OpenAiAnswerService(ILogger<AnswerFactory> logger, RepositoryConfiguratio
 
-            return new AnswerOutput {
-                Answer =  formatted_response, 
-                AnswerType = solution ? "solution" : "suggestion" 
+            return new AnswerOutput
+            {
+                Answer = formatted_response,
+                AnswerType = isSolution ? "solution" : "suggestion"
             };
         }
+        private async Task RetrieveAndAggregateSubqueryResults(
+            IEnumerable<string> subqueries,
+            Dictionary<string, string> labels,
+            HashSet<Issue> uniqueIssues,
+            HashSet<Document> uniqueDocs)
+        {
+            var subqueryTasks = subqueries.Select(async subquery =>
+            {
+                var subqueryIssues = await _ragService.SearchIssuesAsync(_serviceConfiguration.IssueIndexName, _serviceConfiguration.IssueSemanticName, _serviceConfiguration.IssueFieldName, subquery, _serviceConfiguration.Top, _serviceConfiguration.ScoreThreshold, labels);
+                var subqueryDocs = await _ragService.SearchDocumentsAsync(_serviceConfiguration.DocumentIndexName, _serviceConfiguration.DocumentSemanticName, _serviceConfiguration.DocumentFieldName, subquery, _serviceConfiguration.Top, _serviceConfiguration.ScoreThreshold, labels);
+                uniqueIssues.UnionWith(subqueryIssues);
+                uniqueDocs.UnionWith(subqueryDocs);
+            });
+            await Task.WhenAll(subqueryTasks);
+        }
+
+        private Dictionary<string, string> BuildUserPromptData(
+            IEnumerable<Issue> allIssues,
+            IEnumerable<Document> allDocs,
+            IssuePayload issue)
+        {
+            var printableIssues = string.Join("\n\n", allIssues.Select(issue =>
+                $"Title: {issue.Title}\nDescription: {issue.chunk}\nURL: {issue.Url}\nScore: {issue.Score}"));
+
+            var printableDocs = string.Join("\n\n", allDocs.Select(doc =>
+                $"Content: {doc.chunk}\nURL: {doc.Url}\nScore: {doc.Score}"));
+
+            var replacementsUserPrompt = new Dictionary<string, string>
+            {
+                { "Title", issue.Title },
+                { "Description", issue.Body },
+                { "PrintableDocs", printableDocs },
+                { "PrintableIssues", printableIssues }
+            };
+
+            return replacementsUserPrompt;
+        }
+
+        private class IssueIdComparer : IEqualityComparer<Issue>
+        {
+            public bool Equals(Issue x, Issue y) => x.Id == y.Id;
+            public int GetHashCode(Issue obj) => obj.Id.GetHashCode();
+        }
+
+        private class DocumentUrlComparer : IEqualityComparer<Document>
+        {
+            public bool Equals(Document x, Document y) => x.Url == y.Url;
+            public int GetHashCode(Document obj) => obj.Url.GetHashCode();
+        }
+        private record ServiceConfiguration(
```

Comment:
```suggestion

        private record ServiceConfiguration(
```

big nit:  just normalizing spacing.